subprojects/core/src/main/java/org/gradle/caching/internal/tasks/DefaultTaskOutputCachingBuildCacheKeyBuilder.java

```diff
@@ -140,7 +154,10 @@ public boolean isValid() {
 
         @Override
         public String toString() {
-            return String.valueOf(hashCode);
```

Comment:
I came to the same conclusion. We shouldn't have used `toString()` in the first place I guess.